src/averageAges.js

```diff
@@ -56,6 +83,26 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const children = people.filter(person => (
+    (onlyWithSon
+      ? person.sex === 'm'
+      : true
+    )
+    && people.find(mother => mother.name === person.mother)));
+
+  const diffAges = children.map(child => {
+    const childMother = people.find(person => person.name === child.mother);
+
+    return child.born - childMother.born;
+  });
+
+  return averageAges(diffAges);
+}
+
+function averageAges(ages) {
+  const sumAges = ages.reduce((sum, age) => sum + age, 0);
+
+  return sumAges / ages.length || 0;
 }
```

Comment:
You have two identical helpers `averageAges` & `getAverageAge`. Delete one of them please

p.s. `getAverageAge` good variable name instead `averageAges` because you describe what action do yo do in this function.